hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -822,14 +823,17 @@ public static HoodieData<HoodieRecord> convertMetadataToRecordIndexRecords(Hoodi
             if (writeStat.getPath().endsWith(baseFileFormat.getFileExtension())) {
               return BaseFileRecordParsingUtils.generateRLIMetadataHoodieRecordsForBaseFile(basePath, writeStat, writesFileIdEncoding, instantTime, storage);
             } else if (FSUtils.isLogFile(fullFilePath)) {
-              // for logs, we only need to process log files containing deletes
-              if (writeStat.getNumDeletes() > 0) {
-                Set<String> deletedRecordKeys = getRecordKeys(fullFilePath.toString(), dataTableMetaClient,
-                    finalWriterSchemaOpt, maxBufferSize, instantTime, false, true);
-                return deletedRecordKeys.stream().map(recordKey -> HoodieMetadataPayload.createRecordIndexDelete(recordKey)).collect(toList()).iterator();
-              }
-              // ignore log file data blocks.
-              return new ArrayList<HoodieRecord>().iterator();
+              checkState(writeStat instanceof HoodieDeltaWriteStat, "Log file should be associated with a delta write stat");
+              List<String> logFilePaths = ((HoodieDeltaWriteStat) writeStat).getLogFiles().stream()
+                  .map(logFile -> new StoragePath(new StoragePath(dataTableMetaClient.getBasePath(), writeStat.getPartitionPath()), logFile).toString())
+                  .collect(toList());
```

Comment:
also, don't we need to group by fileSlice for all write stats?
for eg, if there are more than 1 log file added to the same file slice, does the current record generation will be right?

Reply:
good point! log files can roll over and we can have more than 1 log file in the same slice. I will update this.